tests/python_functional/src/syslog_ng_config/statements/sources/file_source.py

```diff
@@ -32,8 +33,13 @@ def __init__(self, file_name=None, **options):
         self.driver_name = "file"
         self.group_type = "source"
         self.options = options
-        if file_name:
-            self.positional_option = Path(tc_parameters.WORKING_DIR, file_name)
+        if file_name is not None:
+            if file_name == "":
+                self.positional_option = "''"
+            elif file_name == "auto":
```

Comment:
So how can you create `file("auto");`?
I would rather have an `AUTO_FILENAME = -1` "constant" instead of `"auto"`, which could be handled here (note the type difference, you can still set `"-1"` as the filename, only `-1` as an integer would be handled differently).

Reply:
Thanks for the note, and an idea :) .
For negative testcases it will be good to support -1 (as an integer) also. So I would not reserve it as a special case. It is less chance to choose "auto" as a filename. So I would keep this change ("auto").